api/envoy/type/matcher/regex.proto

```diff
@@ -24,7 +24,10 @@ message RegexMatcher {
     // compiled regex is to evaluate. A regex that has a program size greater than the configured
     // value will fail to compile. In this case, the configured max program size can be increased
     // or the regex can be simplified. If not specified, the default is 100.
-    google.protobuf.UInt32Value max_program_size = 1;
+    //
+    // This field is deprecated; regexp validation should be performed on the management server
+    // instead of being done by each individual client.
+    google.protobuf.UInt32Value max_program_size = 1 [deprecated = true];
```

Comment:
WDYT of leaving this in, but providing a reasonable default max_program_size that is bounded, but can be overridden?

Reply:
In what way is that different from the current field? If `max_program_size` is "not specified, the default is 100."

Reply:
I think I'm getting at what @markdroth is trying to express: this is an API that is intended to be implementable by proxies other than Envoy.

I think maybe he objects to predefining what the default behavior is, and rely solely on the management plane to enforce limits, which concerns me because I am thinking from the perspective of managing a shared-tenancy proxy, and I want the default scenario to consume bounded CPU.

I think maybe we could change the comment from saying "if not specified, the default is 100." to "if not specified, the default limit is implementation-dependent".

Mark, would that be good enough?

Reply:
I don't think changing the default here would address my concern.  The real issue is that we need to be validating this on the management server, not on the client.